RPA worker enters unrecoverable state if `rpa-workspaces` directory is removed

### Describe the Problem

On long-running worker instances, the tmp files might be cleaned up while still in use.

This leads to the worker being in an unrecoverable state, leading to every Job being rejected with:
```JSONL
{"timestamp":"2025-06-19T08:53:25.102941600Z","level":"INFO","message":"Received Job from Zeebe","logger":"io.camunda.rpa.worker.zeebe.ZeebeJobService","thread":"grpc-default-executor-1","context":{"task":"camunda::RPA-Task::default","job":"2251799814758154"},"tags":[]}
{"timestamp":"2025-06-19T08:53:25.102941600Z","level":"INFO","message":"Identified resource key for script","logger":"io.camunda.rpa.worker.zeebe.ZeebeJobService","thread":"grpc-default-executor-1","context":{"resourceKey":"2251799814757016","linkName":"RPAScript"},"tags":[]}
{"timestamp":"2025-06-19T08:53:25.103942500Z","level":"ERROR","message":"Error while executing Job","logger":"io.camunda.rpa.worker.zeebe.ZeebeJobService","thread":"io-2","thrown":{"type":"java.io.UncheckedIOException","message":"java.nio.file.NoSuchFileException: C:\\Users\\RPA\\AppData\\Local\\Temp\\rpa-workspaces16843977839920188756\\workspace7663018471484026582","stacktrace":"java.io.UncheckedIOException: java.nio.file.NoSuchFileException: C:\\Users\\RPA\\AppData\\Local\\Temp\\rpa-workspaces16843977839920188756\\workspace7663018471484026582\r\n\tat io.camunda.rpa.worker.io.DefaultIO.createTempDirectory(DefaultIO.java:175)\r\n\tat io.camunda.rpa.worker.workspace.WorkspaceService.createWorkspace(WorkspaceService.java:45)\r\n\tat io.camunda.rpa.worker.robot.RobotService.lambda$newRobotEnvironment$0(RobotService.java:150)\r\n\tat reactor.core.publisher.MonoSupplier.call(MonoSupplier.java:67)\r\n\tat reactor.core.publisher.FluxSubscribeOnCallable$CallableSubscribeOnSubscription.run(FluxSubscribeOnCallable.java:228)\r\n\tat reactor.core.scheduler.SchedulerTask.call(SchedulerTask.java:68)\r\n\tat reactor.core.scheduler.SchedulerTask.call(SchedulerTask.java:28)\r\n\tat java.base@24.0.1/java.util.concurrent.FutureTask.run(FutureTask.java:328)\r\n\tat java.base@24.0.1/java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:309)\r\n\tat java.base@24.0.1/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1095)\r\n\tat java.base@24.0.1/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:619)\r\n\tat java.base@24.0.1/java.lang.Thread.runWith(Thread.java:1460)\r\n\tat java.base@24.0.1/java.lang.Thread.run(Thread.java:1447)\r\n\tat org.graalvm.nativeimage.builder/com.oracle.svm.core.thread.PlatformThreads.threadStartRoutine(PlatformThreads.java:832)\r\n\tat org.graalvm.nativeimage.builder/com.oracle.svm.core.thread.PlatformThreads.threadStartRoutine(PlatformThreads.java:808)\r\nCaused by: java.nio.file.NoSuchFileException: C:\\Users\\RPA\\AppData\\Local\\Temp\\rpa-workspaces16843977839920188756\\workspace7663018471484026582\r\n\tat java.base@24.0.1/sun.nio.fs.WindowsFileSystemProvider.createDirectory(WindowsFileSystemProvider.java:503)\r\n\tat java.base@24.0.1/java.nio.file.Files.createDirectory(Files.java:647)\r\n\tat java.base@24.0.1/java.nio.file.TempFileHelper.create(TempFileHelper.java:130)\r\n\tat java.base@24.0.1/java.nio.file.TempFileHelper.createTempDirectory(TempFileHelper.java:162)\r\n\tat java.base@24.0.1/java.nio.file.Files.createTempDirectory(Files.java:890)\r\n\tat io.camunda.rpa.worker.io.DefaultIO.createTempDirectory(DefaultIO.java:172)\r\n\t... 14 common frames omitted\r\n"},"context":{"task":"camunda::RPA-Task::default","job":"2251799814758154"},"tags":[]}

```

### Steps to reproduce

- Start RPA worker
- Execute a script
- Find the the `rpa-workspaces*` folder from the logs and delete it manually. This emulates automatic cleanup of tmp files
- Try to execute another script
- It fails


### Expected Behavior

- The worker recovers gracefully from cleanup
 
or

- The worker prevents the deletion of workspaces directory

### Suggested fix

I currently see 2 options to solve this. 

1. Always check if the workspaces folder still exists, an re-create it if necessary
2. Add a `.lock` file in the directory and keep it open during execution. We can prevent accidental deletion of the directory between runs with that

Personally, 1‍⃣ sounds cleaner.


Related to [SUPPORT-27801](https://jira.camunda.com/browse/SUPPORT-27801)

--- a/rpa-worker-application/src/main/java/io/camunda/rpa/worker/workspace/WorkspaceService.java
+++ b/rpa-worker-application/src/main/java/io/camunda/rpa/worker/workspace/WorkspaceService.java
@@ -1,27 +1,38 @@
 package io.camunda.rpa.worker.workspace;
 
 import io.camunda.rpa.worker.io.IO;
+import io.vavr.control.Try;
 import jakarta.annotation.PostConstruct;
+import jakarta.annotation.PreDestroy;
 import lombok.AccessLevel;
 import lombok.Getter;
 import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
 import org.springframework.stereotype.Service;
 
+import java.io.RandomAccessFile;
+import java.nio.channels.FileChannel;
+import java.nio.channels.FileLock;
+import java.nio.channels.OverlappingFileLockException;
 import java.nio.file.Path;
 import java.nio.file.StandardOpenOption;
 import java.util.Collections;
 import java.util.Map;
 import java.util.Optional;
+import java.util.Set;
 import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.CopyOnWriteArraySet;
 import java.util.stream.Stream;
 
 @Service
 @RequiredArgsConstructor
+@Slf4j
 public class WorkspaceService {
 
 	private final IO io;
 
 	private final Map<String, Workspace> workspaces = new ConcurrentHashMap<>();
+	private final Set<LockableFileChannel> locks = new CopyOnWriteArraySet<>();
 	
 	@Getter(AccessLevel.PACKAGE)
 	private Path workspacesDir;
@@ -29,6 +40,7 @@ public class WorkspaceService {
 	@PostConstruct
 	WorkspaceService init() {
 		workspacesDir = io.createTempDirectory("rpa-workspaces");
+		ensureWorkspacesDirExistsAndIsLocked();
 		return this;
 	}
 
@@ -42,6 +54,7 @@ public class WorkspaceService {
 
 
 	public Workspace createWorkspace(String affinityKey, Map<String, Object> properties) {
+		ensureWorkspacesDirExistsAndIsLocked();
 		Path workspacePath = io.createTempDirectory(workspacesDir, "workspace");
 		String workspaceID = workspacePath.getFileName().toString();
 		io.writeString(workspacePath.resolve(".workspace"), workspaceID, StandardOpenOption.CREATE_NEW);
@@ -86,5 +99,32 @@ public class WorkspaceService {
 						io.size(wr.requestedFile()),
 						wr.requestedFile()));
 	}
-
+	
+	private record LockableFileChannel(FileChannel fileChannel, FileLock fileLock) {
+		static LockableFileChannel locking(Path path) {
+			return Try.of(() -> new RandomAccessFile(path.toFile(), "rw"))
+					.mapTry(fh -> new LockableFileChannel(fh.getChannel(), fh.getChannel().lock()))
+					.get();
+		}
+	}
+	
+	private void ensureWorkspacesDirExistsAndIsLocked() {
+		io.createDirectories(workspacesDir);
+		Path lockFile = workspacesDir.resolve(".lock");
+		Try.of(() -> LockableFileChannel.locking(lockFile))
+				.onSuccess(locks::add)
+				.recover(OverlappingFileLockException.class, _ -> null)
+				.onFailure(thrown -> log.atError()
+						.setCause(thrown)
+						.kv("lockFile", lockFile)
+						.log("Couldn't lock workspaces dir"));
+	}
+	
+	@PreDestroy
+	private void destroy() {
+		locks.forEach(lock -> {
+			Try.run(() -> lock.fileLock().release());
+			Try.run(() -> lock.fileChannel().close());
+		});
+	}
 }

